ext/tests/post_hook_type_error.phpt

```diff
@@ -0,0 +1,31 @@
+--TEST--
+Check if type error in post hook is handled
+--EXTENSIONS--
+opentelemetry
+--FILE--
+<?php
+class Foo
+{
+  public function bar(): void
+  {
+    var_dump('bar');
+  }
+}
+
+\OpenTelemetry\Instrumentation\hook(
+    Foo::class,
+    'bar',
+    fn() => var_dump('pre'),
+    fn(string $scope, array $params, mixed $returnValue, ?Throwable $throwable) => var_dump('post')); //NB invalid type for $scope
+
+try {
+  (new Foo())->bar();
+} catch(Exception) {}
+
+var_dump('baz');
```

Comment:
Actually meant inside the `try` block since the exception should already be suppressed in the extension. Or just remove the try/catch block.

Reply:
Come to think of it, `TypeError` does not actually extend `Exception` so the try/catch wouldn't catch it anyway, so it's probably fine as it is.

Reply:
> Come to think of it, TypeError does not actually extend Exception so the try/catch wouldn't catch it anyway, so it's probably fine as it is.

it's cleaner this way. and good news that your fix also seems to apply to non-exception errors